code/getData.py

```diff
@@ -0,0 +1,57 @@
+import tweepy
+from tweepy import OAuthHandler
+import csv
+import time
+
+#API Credentials
+consumer_key = 'GvyzAMpV6G5xsOCf4HtjaKO52'
+consumer_secret = 'VZ8Ra1Vp46D5Gq5Otf6XVCaLzAVmRPHGG4MLHXpKSO7zWZJFPO'
+access_token = '804568727685447680-Wuv4RjqRvkp1OoAXeIVJLokl4boO1Hl'
+access_secret = 'HBcA0FJ8YBbg9SXGLy8JhdF2PKpGjyUeIvgZRBWzAZurZ'
+
+auth = OAuthHandler(consumer_key, consumer_secret)
+auth.set_access_token(access_token, access_secret)
+
+api = tweepy.API(auth)
+
+# Open/create a file to write data to
+csvFile = open('result.csv','w')
+
+#Use csv writer
+csvWriter = csv.writer(csvFile)
+
+words = ['Asleep and never wake','Just want to sleep forever','Take my own life','Can’t do this anymore','Kill myself','Thoughts of suicide'
+,'Could just fall asleep','Killing myself','Tired of being alone',
+'Die in my sleep','Life is so meaningless','Tired of being lonely',
+'Don’t want to be here','Life is too hard','To end this nightmare',
+'Don’t want to exist','Life is worthless','To hurt myself',
+'Don’t want to go on','My death would','To live anymore',
+'Don’t want to live','My life consists of nothing','Want it to be over',
+'Don’t want to try anymore','My life is pointless','Want to be alive anymore',
+'Don’t want to wake up','My life is this miserable','Want to be around anymore',
+'End it all','My life isn’t worth','Want to be dead',
+'End my life','Not want to be alive','Want to be gone',
+'End this pain','Nothing to live for','Want to be here anymore',
+'Ending it all','Point in living','Want to die',
+'Hate my life','Put an end to this','Want to disappear',
+'Hate myself','Ready to die','Want to end it',
+'I’m drowning','Really need to die','Wanted to die',
+'I’m leaving now','Stop the pain','Wanting to kill yourself',
+'I’m worthless','Suicidal','What is wrong with me',
+'Isn’t worth living','Suicide','Why should I continue living',
+'Just want to give up','Take it anymore']
+
```

Comment:
https://github.com/IE-NITK/TwitterSuicidalAnalysis/blob/6acfcf4d7d485ab29422392120104fe97d3b44e3/code/getData.py#L44-L57

I think I know why your program takes long intervals. It would be better if you collect tweets and then filter out, something like this:

```suggestion
# stdOutListener(.) prints received tweets to stdout (use `>> filename` to capture the stdout) 
stream = Stream(oAuthHandler, stdOutListener)
stream.filter(track=words)
```

Reply:
What i have understood about stream is that it only gets tweets which are posted while the program is running and not tweets posted from before

Reply:
However, I will try this also later on today and tell you what happens by tonight

Reply:
Yeah that's true. But we are slowing down the process even more, aren't we? Say, twitter receives the following tweet when we are searching with word=`Tired of being alone`:
`I want to die in the arms of my love :(`
What is done? - we discard it, right? And probably lose it for now and recollect it while searching for that particular word! So, isn't it performing the same task multiple times?

Reply:
If you've already mined the tweets, leave it be. Don't change anything. As long as it works, for now time shouldn't be an issue 🙃

Reply:
Yeah, so i don't think streaming is advisable, whereas in the normal search, it looks for all the tweets with the "word" in the last month and we can directly add them to our file.

Reply:
In that case, you can use `Twitter Search API` rather than `Streaming API` or `Firehose API`. It's upto you. Decide and make a smart move. The problem with old data is that our keyword search might have to be refined. Anyways just try to mine as much as you can :)